### PR TITLE
fix: fix `page_options` for `PlaywrightBrowserPlugin`

### PR DESCRIPTION
**Pull request by Mantisus**
### Description

- fix  `page_options` for `PlaywrightBrowserPlugin`

### Issues

- Closes: #755, #751 

### Testing

- Add test for check workability `page_options` in `PlaywrightBrowserPlugin`

### Checklist

- [ ] CI passed

### CODE REVIEW

**Comment by Mantisus** [comment]
My initial thought was to change the name of `page_options` to `context_options`. However, I think this might confuse users who are not so familiar with `playwright`. Especially considering that the parameters taken by `new_context` and `new_page` in `playwright` are identical.

So I've updated the docstring to explicitly state that these parameters are passed when the context is created.

File: src/crawlee/browsers/_playwright_browser_controller.py [hunk 1]
```diff
@@ -130,17 +129,20 @@ def _on_page_close(self, page: Page) -> None:
...
-    async def _create_browser_context(self, proxy_info: ProxyInfo | None = None) -> BrowserContext:
+    async def _create_browser_context(
+        self, page_options: Mapping[str, Any] | None = None, proxy_info: ProxyInfo | None = None
+    ) -> BrowserContext:
         """Create a new browser context with the specified proxy settings."""
         if self._header_generator:
             common_headers = self._header_generator.get_common_headers()
             sec_ch_ua_headers = self._header_generator.get_sec_ch_ua_headers(browser_type=self.browser_type)
             user_agent_header = self._header_generator.get_user_agent_header(browser_type=self.browser_type)
             extra_http_headers = dict(common_headers | sec_ch_ua_headers | user_agent_header)
```
Comment (Pijukatel) [inline src/crawlee/browsers/_playwright_browser_controller.py:140]: I think, that this extra_http_headers will never be used with current change. So either remove it or do some logic that decides when to use page_options['extra_http_headers'] and when extra_http_headers.

File: src/crawlee/browsers/_playwright_browser_controller.py [hunk 2]
```diff
@@ -130,17 +129,20 @@ def _on_page_close(self, page: Page) -> None:
...
-            user_agent = user_agent_header.get('User-Agent')
```
Comment (Pijukatel) [inline src/crawlee/browsers/_playwright_browser_controller.py:140]: Just a question: Is user_agent expected to be part of page_options? Or why is it removed here?

File: src/crawlee/browsers/_playwright_browser_controller.py [hunk 3]
```diff
@@ -98,13 +98,12 @@ async def new_page(
...
         if not self._browser_context:
-            self._browser_context = await self._create_browser_context(proxy_info)
+            self._browser_context = await self._create_browser_context(page_options, proxy_info)
 
         if not self.has_free_capacity:
             raise ValueError('Cannot open more pages in this browser.')
 
-        page_options = dict(page_options) if page_options else {}
-        page = await self._browser_context.new_page(**page_options)
+        page = await self._browser_context.new_page()
```
Comment (Pijukatel) [inline src/crawlee/browsers/_playwright_browser_controller.py:106]: Another question: Previously it was possible to pass arguments to self._browser_context.new_page(), now it is not possible. Was that not a valid use case?

File: src/crawlee/browsers/_playwright_browser_controller.py (hunk 2 above)
Comment (Mantisus) [reply src/crawlee/browsers/_playwright_browser_controller.py:140]: user-agent is already set in `extra_http_headers` from `user_agent_header` headers, there is no need to set it twice.

File: src/crawlee/browsers/_playwright_browser_controller.py (hunk 3 above)
Comment (Mantisus) [reply src/crawlee/browsers/_playwright_browser_controller.py:106]: Earlier to create a page we used to use [`browser.new_page()`](https://playwright.dev/python/docs/api/class-browser#browser-new-page) method which takes page parameters. But some time ago we switched to using context and we create a page using [`browser_context.new_page()`](https://playwright.dev/python/docs/api/class-browsercontext#browser-context-new-page) which doesn't take any parameters. Now the page opening parameters must be set when creating the context

File: tests/unit/browsers/test_browser_pool.py [hunk 4]
```diff
@@ -151,3 +151,12 @@ async def test_methods_raise_error_when_not_active() -> None:
 
     async with browser_pool:
         assert browser_pool.active is True
+
+
+async def test_with_plugin_contains_page_options(httpbin: URL) -> None:
+    plugin = PlaywrightBrowserPlugin(page_options={'user_agent': 'My Best Uset-Agent'})
```
Comment (Pijukatel) [inline tests/unit/browsers/test_browser_pool.py:157]: Super tiny irrelevant typo: My Best Use**t**-Agent

Comment (Mantisus) [reply tests/unit/browsers/test_browser_pool.py]: Thank you for noticing that!

**Review by Pijukatel — APPROVED** [review]

File: src/crawlee/browsers/_playwright_browser_controller.py (hunk 2 above)
Comment (vdusek) [inline src/crawlee/browsers/_playwright_browser_controller.py:140]: This was here to replace the default "headless"-something user agent. Why is it removed? Does it mean we're now using the default (headless) user agent?

Comment (Mantisus) [reply src/crawlee/browsers/_playwright_browser_controller.py:140]: We have a similar test

https://github.com/apify/crawlee-python/blob/master/tests/unit/playwright_crawler/test_playwright_crawler.py#L106

**Pull request merged**